ebb_lint/flake8.py

```diff
@@ -96,10 +97,14 @@ def find_comments(s, base_byte=0):
 
 
 def read_file_using_source_encoding(filename):
+    if filename == 'stdin':
+        reader = codecs.getreader(getattr(sys.stdin, 'encoding') or 'utf-8')
+        return reader(sys.stdin).read()
+
     with open(filename, 'rb') as infile:
         encoding = tokenize.detect_encoding(infile.readline)[0]
-    with io.open(filename, 'r', encoding=encoding) as infile:
-        return infile.read()
+    with io.open(filename, 'r', encoding=encoding) as infile_with_encoding:
+        return infile_with_encoding.read()
```

Comment:
Ugh, python 3.

I wonder if it would be simpler to depend on click for `click.open_file`, since it can handle getting bytes out of stdin.